src/hcaptcha-script.js

```diff
@@ -0,0 +1,74 @@
+/**
+ * Async hcaptcha api.js loader.
+ *
+ * It makes sure `endpoint` is loaded ONCE on the page despite calling load() multiple times.
+ *
+ * Usage:
+ * 1. import hcaptchaScript from './hcaptcha-script';
+ * 2. when web component is mounted do:
+ *      hcaptchaScript
+ *        .load('endpoint', 'language', 'reCaptchaCompat')
+ *        .then(() => console.log('hcaptcha is loaded so it is safe to be used'))
+ *        .catch((err) => console.error('failed to load the hcaptcha', err));
+ */
+
+/**
+ * Used to prevent loading api.js multiple times
+ * @type {string}
+ */
+const SCRIPT_ID = 'hcaptcha-api-script-id';
+
+const HCAPTCHA_LOAD_FN_NAME = '_hcaptchaOnLoad';
+
+class HcaptchaScript {
```

Comment:
Why make this a class, couldn't this be an async function?

Reply:
Sure thing. The point was to have a single instance of this so we can resolve promise once. Please check out the last commit.